### DISCUSSION TITLE
Next JS returning 500 error after adding new product

### DISCUSSION THREAD

**Discussion by fatdoglabs** [vercel/next.js#73216] [Help]
### Summary

What are the details of your problem ?
I am encountering an issue in my ```nextjs``` application, The project i am working on is an e-commerce application where backend framework used is ```laravel``` and the frontend framework is nextjs.The application is currently hosted on **Digital Ocean App Platform** and has two separate environments, i.e `production` and `staging`.When i add a new product in admin panel, the new product is created and hence it is shown in homepage, but the problem is when i navigate to products detail page,and also when i hover on the product on homepage which uses nextjs `Link` component and has `prefetch` enabled it gives me 500 internal server error. 

Things to note: 
---
1. I tested locally using both staging and production environments to try to replicate the issue , but on local machine it is working.
2. I tested on staging environment that is deployed on **Digital Ocean App Platform** by adding a new product, and it is also working fine.
3. This issue didn't exist (or i didn't know it existed) until 3 weeks ago.
4. The application has currently around 1450 pages (**ISR**) on production environment.
---
Application specification and server specification 
---
Nextjs and react version used:
```json
{
 "next": "13.0.6",
 "react": "18.2.0",
 "react-dom": "18.2.0",
}
```
Production:
- RAM:4GB
- CPU:4vCPU's
- Bandwidth: 250GB
- Instance:2

Staging:
- RAM:2GB
- CPU:2vCPU's
- Bandwidth: 100GB
- Instance:1

Local Machine
- RAM:32GB
- CPU: Intel i5 13th gen

Below is the code for both homepage and product detail page : 

Get static props for home page:
```js
export async function getStaticProps() {
    const queryClient = new QueryClient();

    await queryClient.prefetchQuery({
        queryKey: [KEYS.bannerList],
        queryFn: getBannerList,
    });
    await queryClient.prefetchQuery({
        queryKey: [KEYS.featuredCategory],
        queryFn: fetchFeaturedCategory,
    });
    await queryClient.prefetchQuery({
        queryKey: [KEYS.getAllHomeData],
        queryFn: fetchAllHomeData,
    });

    return {
        props: {
            dehydratedState: dehydrate(queryClient),
        },
        revalidate: 10
    };
}
```
Get static paths and getstatic props for product detail page:

```js
export async function getStaticPaths() {
    const data = await fetch(`${baseURL}/api/all-product`, {
        cache: 'no-cache',
    })
        .then((response) => response.json())
        .then((data) => data?.data);

    const paths = await data?.map((product) => ({
        params: { product_slug: product.slug },
    }));

    return {
        paths: paths || [],
        fallback: 'blocking',
    };
}

export async function getStaticProps(context) {
    const { product_slug } = context.params;
    const queryClient = new QueryClient();

    await queryClient.prefetchQuery({
        queryKey: [KEYS.productsDetails, { product_slug }],
        queryFn: () => getProductDetails(product_slug),
    })
    return {
        props: {
            dehydratedState: dehydrate(queryClient),
        },
        revalidate: 10
    };
}
```
These are the general info's that i get back from network tabs:
*GENERAL*
- Request URL:{{hostname}}/_next/data/7CjVi7i3xpZZ4UAXQAoPQ/product/asus-vivobook-s-14-flip-2024-ryzen5-price.json
- Request Method: HEAD
- Status Code:500 Internal Server Error
- Referrer Policy:no-referrer

*RESPONSE HEADERS*
- alt-svc:h3=":443"; ma=86400
- cache-control:private
- cf-cache-status:MISS
- cf-ray:8e8b0f2acc1d7f25-MAA
- content-type:text/html; charset=utf-8
- date:Tue, 26 Nov 2024 15:54:50 GMT
- priority:u=1,i
- server:cloudflare
- server-timing:cfExtPri
- vary:Accept-Encoding
- x-do-app-origin:ceec05bb-dd58-4262-bbe3-222ad9cedc15
- x-do-orig-status:500
- x-nextjs-matched-path:/product/[product_slug]
- x-powered-by:Next.js

- I tried to replicate the issue in local environment in which i had no luck , so my best option is to go to digital ocean's dashboard and **force rebuild and redeploy**
which only works until the new product is added 
![image](https://github.com/user-attachments/assets/0a6396f5-aedc-4387-83f7-1efa1cf636f4)

- I also tried by updating the version to `nextjs` to latest stable version.
- I tried using nextjs **standalone** output mode in local environment using docker, which is also working fine.

**Comment by denis23x** [comment]
Hi @fatdoglabs 

Why are you sure that the error is in the Next? Maybe you have an error in Laravel and because of it the error is already happening on the frontend.

Try to hit problem product endpoint via Postman

- If 200, compare the answer with other products and find the differences that lead to the error
- If 500, there is a bug in Laravel and you should ask the question on the appropriate forum

**Reply by fatdoglabs** [reply]
Hi @denis23x 

Thanks for replying, i did check it from my side to identify the problem  and it seems it is working fine from API side. The thing is this specific problem only arrises on production environment .
the runtime logs shows this message :
```
Warning: data for page "/product/[product_slug]" (path "/product/test-product-df4534545") is 789 kB which exceeds the threshold of 128 kB, this amount of data can reduce performance.
See more info here: https://nextjs.org/docs/messages/large-page-data
```
Although i am not sure that the error is in Next, The request that is going to 
`Request URL:{{hostname}}/_next/data/7CjVi7i3xpZZ4UAXQAoPQ/product/est-product-df4534545.json` is  throwing 500 error . 

NOTE: This is the latest product that i created for test purpose.

I would like to know if there is any other method to try to debug this issue.

Thank you.

**Reply by denis23x** [reply]
Not sure will it helpful, it's just a warning 🤷‍♂️

``` javascript
const nextConfig = {
  experimental: {
    largePageDataBytes: 800 * 1000, // 800kb
  },
};
```
